packages/opentelemetry-node/src/instrumentation/utils.ts

```diff
@@ -72,3 +72,30 @@ export function isSupportedVersion(
 export function searchPathForTest(searchPath: string) {
   module.paths.push(searchPath);
 }
+
+// Includes support for npm '@org/name' packages
+// Regex: .*?node_modules(?!.*node_modules)\/(@[^\/]*\/[^\/]*|[^\/]*).*
+// Tests: https://regex101.com/r/lW2bE3/6
+const moduleRegex = new RegExp(
```

Comment:
This seems like a really dangerous way to handle this. Why do we not just do `require.resolve(name)` to get the full path, then check if it is in the require cache?

Reply:
I don't mind using `require.solve` however i don't see in which case that regex is dangerous, could you explain why ?

Reply:
Just that this is a very complex regex that would be difficult to debug if there was an issue

Reply:
Updated !